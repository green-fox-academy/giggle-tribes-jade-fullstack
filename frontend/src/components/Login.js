import React from 'react';

const Login = () => {

    return (
        <p>Login screen.</p>
    );
};

export default Login;
