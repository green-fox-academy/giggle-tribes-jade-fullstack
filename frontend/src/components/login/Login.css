body {
  background-image: url('/bg3-01.png');
  background-color: black;
  background-position: center;
}

.login {
  display: flex;
  flex-direction: column;
}

.title h2 {
  text-align: center;
  font-size: 4rem;
  margin-top: 3%;
  margin-bottom: 0.2%;
  color: whitesmoke;
  justify-content: center;
}

.form {
  background-color: rgba(169, 169, 169, 0.842);
  margin: 3.15rem auto;
  width: 31.5rem;
  border-radius: 3px;
  padding-left: 1.5rem;
  padding-right: 2rem;
  padding-bottom: 3.5rem;
  justify-content: center;
}

.registrationButton {
  width: 9rem;
  height: 3.4rem;
  background-color: green;
  border: 0px;
  border-radius: 4px;
  color: white;
  font-size: 1.25rem;
  letter-spacing: 1px;
  margin-top: 3.25rem;
  box-shadow: 0px 3px 5px #9b9b9b;
  justify-content: flex-end;
}

.submitLine {
  display: flex;
  justify-content: flex-end;
}

.errorMessage {
  color: red;
  font-size: 1.25rem;
  border: 0px;
  border-radius: 4px;
  font-size: 1.25rem;
  letter-spacing: 1px;
  margin-top: 4.25rem;
}

.registrationButton:hover {
  background-color: #75be3e;
}

.registrationButton:focus {
  outline: none;
}

input {
  border-bottom: 3px solid green;
  width: 100%;
  margin-top: 1.8rem;
  height: 3.5rem;
  font-size: 1.6rem;
  border-left: 0px solid lightgrey;
  border-top: 0px solid lightgrey;
  border-right: 0px solid lightgrey;
  background-color: rgba(243, 241, 239, 0);
  outline: none;
}

.red {
  border-bottom: 3px solid red;
}

.redimage {
  display: flex;
  justify-content: space-evenly;
  background-image: url('/alert_sign.png');
  background-repeat: no-repeat;
  color: red;
}

.alertnotext {
  font-size: 0px;
}

.green {
  border-bottom: 3px solid green;
}
