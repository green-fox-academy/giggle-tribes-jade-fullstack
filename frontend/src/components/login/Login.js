import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import './Login.css';
import { generalFetch } from '../../services/fetchService';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleUsername = this.handleUsername.bind(this);
    this.handlePassword = this.handlePassword.bind(this);
  }

  handleSubmit(event) {
    event.preventDefault();

    let userData = this.state;
    const { history } = this.props;

    generalFetch('sessions', {
      method: 'POST',
      body: JSON.stringify(userData),
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
    })
      .then(data => {
        if (data.error) {
          return false;
        } else {
          localStorage.setItem('TRIBES_TOKEN', data.token);
          localStorage.setItem('kingdom', parseInt(data.kingdomId));
          return true;
        }
      })
      .then(success => {
        if (success) {
          history.push('/kingdom/buildings');
        }
      });
  }

  handleUsername(e) {
    let value = e.target.value;
    this.setState(prevState => ({ ...prevState, username: value }));
  }

  handlePassword(e) {
    let value = e.target.value;
    this.setState(prevState => ({ ...prevState, password: value }));
  }

  render() {
    return (
      <div className="login">
        <div className="title">
          <h2>Tribes of Vulpes</h2>
        </div>

        <div className="form">
          <form onSubmit={this.handleSubmit}>
            <div>
              <input
                type="text"
                name="username"
                placeholder="Username"
                onChange={this.handleUsername}
              ></input>
            </div>

            <div>
              <input
                type="password"
                name="password"
                placeholder="Password"
                onChange={this.handlePassword}
              ></input>
            </div>

            <div className="submitLine">
              {this.props.error && (
                <div className="errorMessage">{this.props.error}</div>
              )}
              <button type="submit" className="loginButton">
                SIGN IN
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default withRouter(Login);
