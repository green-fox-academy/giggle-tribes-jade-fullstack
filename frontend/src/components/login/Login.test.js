import React from 'react';
import { Router as Router, Route} from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Login from './Login';
import { createMemoryHistory } from 'history';
import { render, fireEvent } from '@testing-library/react';

const history = createMemoryHistory({
    initialEntries: ['/']
  });
  
let container;

const mockStore = configureStore([]);
const mockedStore = mockStore({});

const login = renderer.create(
    <Router history={history}>
        <Route path = '/'>
            <Provider store={mockedStore}>
              <Login />
            </Provider>
        </Route>
    </Router>,
    container
  )

  let utils = ()=>{return render(
    <Router history={history}>
        <Route path = '/'>
          <Provider store={mockedStore}>
            <Login />
          </Provider>
        </Route>
    </Router>,
    container
);}

  it('renders correctly', () => {
    mockedStore.dispatch = jest.fn(); 
    login.toJSON();
    expect(login).toMatchSnapshot();
  });


  let fieldByPlaceholderName = (placeholder) => {return utils().getByPlaceholderText(placeholder)}
  
  it('either input field empty shows error message', async() => {
    let inputfield = fieldByPlaceholderName("Username" || "Password")
    fireEvent.change(inputfield, {target: {value: ""}})
    const text = utils(/required/i);
    expect(text.baseElement).toBeInTheDocument();
  });

  it('both input field empty shows error message', async() => {
    let inputfield = fieldByPlaceholderName("Username" && "Password")
    fireEvent.change(inputfield, {target: {value: ""}})
    const text = utils(/required/i);
    expect(text.baseElement).toBeInTheDocument();
  });
