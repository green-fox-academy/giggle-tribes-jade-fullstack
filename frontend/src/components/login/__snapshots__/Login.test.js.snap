// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="login"
>
  <div
    className="title"
  >
    <h2>
      Tribes of Vulpes
    </h2>
  </div>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <input
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
      </div>
      <div>
        <input
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="submitLine"
      >
        <button
          className="loginButton"
          type="submit"
        >
          SIGN IN
        </button>
      </div>
    </form>
  </div>
</div>
`;
